MPDZ.py

```diff
@@ -0,0 +1,471 @@
+        if sep:
+            threel_form = threel_form[:-1]
+        res.append(threel_form)
+    return res
+
+
+coperations = {
+    "length": length_info,
+    "percentage": count_percentage_aa,
+    "pattern": find_pattern,
+    "3Letter_name": rename_three_letter_name,
+    "DNA_code": get_protein_gene,
+}
+
+
+retrnaslation_dict = {
+    "F": "TTC",
+    "f": "ttc",
+    "L": "TTA",
+    "l": "tta",
+    "S": "TCG",
+    "s": "tcg",
+    "Y": "TAC",
+    "y": "tac",
+    "C": "TGC",
+    "c": "tgc",
+    "W": "TGG",
+    "w": "tgg",
+    "P": "CCC",
+    "p": "ccc",
+    "H": "CAT",
+    "h": "cat",
+    "Q": "GAA",
+    "q": "gaa",
+    "R": "CGA",
+    "r": "cga",
+    "I": "ATT",
+    "i": "att",
+    "M": "ATG",
+    "m": "atg",
+    "T": "ACC",
+    "t": "acc",
+    "N": "AAT",
+    "n": "aat",
+    "K": "AAA",
+    "k": "aaa",
+    "V": "GTT",
+    "v": "gtt",
+    "A": "GCA",
+    "a": "gca",
+    "D": "GAT",
+    "d": "gca",
+    "E": "GAG",
+    "e": "gag",
+    "G": "GGG",
+    "g": "ggg",
+}
+
+
+threel = {
+    "A": "ALA",
+    "R": "ARG",
+    "N": "ASN",
+    "D": "ASP",
+    "V": "VAL",
+    "H": "HIS",
+    "G": "GLY",
+    "Q": "GLN",
+    "E": "GLU",
+    "I": "ILE",
+    "L": "LEU",
+    "K": "LYS",
+    "M": "MET",
+    "P": "PRO",
+    "S": "SER",
+    "Y": "TYR",
+    "T": "THR",
+    "W": "TRP",
+    "F": "PHE",
+    "C": "CYS",
+    "a": "ala",
+    "r": "arg",
+    "n": "asn",
+    "d": "asp",
+    "v": "val",
+    "h": "his",
+    "g": "gly",
+    "q": "gln",
+    "e": "glu",
+    "i": "ile",
+    "l": "leu",
+    "k": "lys",
+    "m": "met",
+    "p": "pro",
+    "s": "ser",
+    "y": "tyr",
+    "t": "thr",
+    "w": "trp",
+    "f": "phe",
+    "c": "cys",
+}
+
+
+def protein_tool(*proteins, options=None):
+    proteins = list(proteins)
+
+    operations = {
+        "compare": compare,
+        "length": length_info,
+        "percentage": count_percentage_aa,
+        "pattern": find_pattern,
+        "3Letter_name": rename_three_letter_name,
+        "DNA_code": get_protein_gene,
+    }
+
+    if options == "compare":
+        result = operations[options](proteins[:-2], proteins[-2], proteins[-1])
+        return result
+    elif options == "3Letter_name":
+        result = operations[options](proteins[:-1], proteins[-1])
+        return result
+    elif options == "length" or options == "percentage" or options == "DNA_code":
+        result = []
+        for protein in proteins:
+            res = operations[options](protein)
+            result.append(res)
+        return result
+    else:
+        raise ValueError("Incorrect options input, please try again")
+
+
+from Bio import SeqIO
+#from Bio.SeqUtils import GC
+
+class FastQFilter:
+    def __init__(self, input_file, output_file, min_length, min_quality, min_gc):
+        self.input_file = input_file
+        self.output_file = output_file
+        self.min_length = min_length
+        self.min_quality = min_quality
+        self.min_gc = min_gc

```

Comment:
Длина и GC-состав, по условию задачи, задаются двумя границами (в функциональной реализации для домашки №5, в коде выше, это учтено) и имеют значения аргументов по умолчанию:

```suggestion
from typing import Tuple, Union

class FastQFilter:
    def __init__(self,
                 input_file: str,
                 output_file: str = None,
                 length_bounds: Tuple[int, int] = (0, 2**32),
                 quality: int = 0,
                 gc_bounds: Union[Tuple[int, int], int] = (0, 100)):
        self.input_file = input_file
        self.output_file = output_file
        self.min_length, self.max_length = length_bounds
        self.quality = quality
        self.min_gc, self.max_gc = gc_bounds
```